src/core/Microsoft.PowerToys.Settings.UI/Converters/ModuleEnabledToForegroundConverter.cs

```diff
@@ -15,13 +12,56 @@ public sealed class ModuleEnabledToForegroundConverter : IValueConverter
         public object Convert(object value, Type targetType, object parameter, string language)
         {
             bool isEnabled = (bool)value;
-            if (isEnabled)
+            GeneralSettings generalSettings = SettingsUtils.GetSettings<GeneralSettings>(string.Empty);
+
+            var defaultTheme = new Windows.UI.ViewManagement.UISettings();
+            var uiTheme = defaultTheme.GetColorValue(Windows.UI.ViewManagement.UIColorType.Background).ToString();
+
+            string selectedTheme = generalSettings.Theme.ToLower();
+
+            if (selectedTheme == "dark")
+            {
+                if (isEnabled)
+                {
+                    return (SolidColorBrush)Application.Current.Resources["DarkForegroundBrush"];
+                }
+                else
+                {
+                    return (SolidColorBrush)Application.Current.Resources["DarkForegroundDisabledBrush"];
+                }
+            }
+            else if (selectedTheme == "light")
+            {
+                if (isEnabled)
+                {
+                    return (SolidColorBrush)Application.Current.Resources["LightForegroundBrush"];
+                }
+                else
+                {
+                    return (SolidColorBrush)Application.Current.Resources["LightForegroundDisabledBrush"];
+                }
+            }
+            else if (selectedTheme == "system" && uiTheme == "#FF000000")
```

Comment:
Should be merged with "dark" block above? Seems to be the same? 
So "dark" || "system" && uitheme ... is using dark brushes, everything else is using light ones right? 
Therefore we only need one top level if/else in this converter. 
I would even argue that we should have multiple resource dictionaries which calls colors by the same name, such as "primaryTextBrush", "secondary..." and we can have multiple of those resouce dictionaries such as "lightColors", "darkColors"... then we only need some top level class that would switch those resource dictionaries - that is generally the way to support themes - views/converters should not care about the current theme. If this change would be big, let just open a new issue and I can have a look into that - if we have that we wouldnt need those converters at all.

Reply:
I agree, will make the statement change.

Yeah, current theme setting is a bit messy atm. Since we can't change the App theme at runtime (only in the constructor) we're running into these kinds of issues.

Maybe it would be best to just restart the settings app and change the App theme. That would solve this issue, we'd have cleaner theme implementation overall and we'd fix other minor bugs (like tooltips not changing correctly when switching themes). Cc @laviusmotileng

Reply:
To clarify: the reason why we need this if statement in the converter is because we were calling the right brushes. These would normally switch automatically (and actually do when switching themes from the W10 Settings app) form e.g. black to white. Just not when switching via the general page, since we're not overriding the App theme but the requesttheme of the shellpage.

Reply:
Agree, I would say that restarting settings app is not that bad when changing theme and would make it cleaner code wise.

Reply:
But we shouldn't restart automatically.

We should show an hint (like for the admin thing) to the user: `Please restart setting to apply`